fmpz_poly/eulerian_polynomial.c

```diff
@@ -0,0 +1,60 @@
+/*
+    Copyright (C) 2021 Albin Ahlbäck
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include "fmpz_poly.h"
+
+void
+fmpz_poly_eulerian_polynomial(fmpz_poly_t poly, const ulong n)
+{
+    ulong nx, kx;
+
+    /* A_0, A_1 = 1 */
+    if (n <= 1)
+    {
+        fmpz_poly_one(poly);
+        return;
+    }
+
+    /* Ensure that we have place for coefficients */
+    fmpz_poly_fit_length(poly, n);
```

Comment:
Am I doing the right thing here? I want to make sure that we have enough coefficients and that the coefficients for `x^n, x^{n + 1}, ...` is zero.